modulemd/tests/test-modulemd-modulestream.c

```diff
@@ -331,6 +331,153 @@ module_stream_v2_test_documentation (ModuleStreamFixture *fixture,
   g_clear_object (&stream);
 }
 
+static void
+module_stream_v1_test_components (ModuleStreamFixture *fixture,
+                                  gconstpointer user_data)
+{
+  g_autoptr (ModulemdModuleStreamV1) stream = NULL;
+  g_autoptr (ModulemdComponentRpm) rpm_component = NULL;
+  g_autoptr (ModulemdComponentModule) module_component = NULL;
+  ModulemdComponent *retrieved_component = NULL;
+  g_autofree gchar *component_name = NULL;
+  g_auto (GStrv) component_names = NULL;
+
+  stream = modulemd_module_stream_v1_new (NULL, NULL);
+
+  // Add a RPM component to a stream
+  rpm_component = modulemd_component_rpm_new ("rpmcomponent");
+  modulemd_module_stream_v1_add_component (stream,
+                                           (ModulemdComponent *)rpm_component);
+  component_names =
+    modulemd_module_stream_v1_get_rpm_component_names_as_strv (stream);
+  g_assert_true (
+    g_strv_contains ((const gchar *const *)component_names, "rpmcomponent"));
+
+  retrieved_component =
+    (ModulemdComponent *)modulemd_module_stream_v1_get_rpm_component (
+      stream, "rpmcomponent");
+  g_assert_nonnull (retrieved_component);
+  g_object_get (retrieved_component, "name", &component_name, NULL);
+  g_assert_cmpstr (component_name, ==, "rpmcomponent");
```

Comment:
Instead of comparing the component name, just use `modulemd_component_equals()` to ensure that you get back a component that is equivalent to `rpm_component`.